### PR TITLE
State: Introduce selectors for whether Jetpack modules require connection

### PR DESCRIPTION
**Pull request by tyxla**
This PR introduces 2 new selectors that we'll use to get all Jetpack modules that require connection, and whether a certain module requires connection. We'll need them in order to be able to disable certain functionality, like we do in the Jetpack settings page. see https://github.com/Automattic/jetpack/pull/6180 for the approach in Jetpack:

Those 2 selectors are similar to the ones in Jetpack:
* `getJetpackModulesRequiringConnection()` - retrieve all modules that require connection
* `isJetpackModuleUnavailableInDevelopmentMode()` - whether a certain module is unavailable in dev mode

Once we have these selectors in place, we'll be able to implement nudges and disable certain UI pieces in Jetpack Settings cards, following the example of https://github.com/Automattic/jetpack/pull/6180.

To test:
* Checkout this branch
* Verify all tests pass:

```
npm run test-client client/state/selectors/test/get-jetpack-modules-requiring-connection.js 
```

```
npm run test-client client/state/selectors/test/is-jetpack-module-unavailable-in-development-mode.js
```

### CODE REVIEW

File: client/state/selectors/index.js [hunk 1]
```diff
@@ -60,6 +61,7 @@ export isDeactivatingJetpackModule from './is-deactivating-jetpack-module';
 export isDomainOnlySite from './is-domain-only-site';
 export isFetchingJetpackModules from './is-fetching-jetpack-modules';
 export isJetpackModuleActive from './is-jetpack-module-active';
+export isJetpackModuleUnavailableInDevelopmentMode from './is-jetpack-module-unavailable-in-development-mode';
```
Comment (oskosk) [inline client/state/selectors/index.js]: What's the use case where the selector  `isJetpackModuleUnavailableInDevelopmentMode` would be more practical than `isJetpackModuleAvailableInDevelopmentMode` ?

Comment (tyxla) [inline client/state/selectors/index.js]: First, this is copying from the functionality in https://github.com/Automattic/jetpack/pull/6180, which uses the same pattern. Second, in most cases that we'll need to implement this selector, we'll need to check against the negative case, for example:

* We'll display a nudge that the site is in development mode only if the module is unavailable in dev mode
* We'll disable the form controls within a module only if the module is unavailable in dev mode.

That said, if we make it `isJetpackModuleAvailableInDevelopmentMode`, we'll have to use it `! isJetpackModuleAvailableInDevelopmentMode` everywhere, so it makes sense for it to check against the negative case.

**Comment by tyxla** [comment]
@oskosk got any other finds on this one?

**Comment by oskosk** [comment]
Nope. 🚢

**Pull request merged**